pubkey/src/lib.rs

```diff
@@ -162,6 +162,110 @@ impl From<u64> for PubkeyError {
 #[cfg_attr(feature = "dev-context-only-utils", derive(Arbitrary))]
 pub struct Pubkey(pub(crate) [u8; 32]);
 
+#[cfg(all(feature = "rand", not(target_os = "solana")))]
+mod hasher {
+    use rand::Rng;
+
+    /// A special Hasher specifically for Pubkeys. Uses random 8 bytes as
+    /// input. Since pubkeys are effectively random, hashing them is pure overhead.
+    /// Use with care, this may be abusable.
+    #[derive(Default)]
+    pub struct PubkeyHasher {
+        offset: usize,
+        state: u64,
+    }
+
+    impl std::hash::Hasher for PubkeyHasher {
+        #[inline]
+        fn finish(&self) -> u64 {
+            self.state
+        }
+        #[inline]
+        fn write(&mut self, bytes: &[u8]) {
+            let chunk: &[u8; 8] = bytes[self.offset..self.offset + 8].try_into().unwrap();
+            self.state = u64::from_ne_bytes(*chunk);
+        }
+    }
+
+    /// A special HasherBuilder specifically for Pubkeys. Uses random 8 byte slice as
+    /// input. Since pubkeys are effectively random, hashing them is pure overhead.
+    /// Use with care, this is not a secure hasher!
+    #[derive(Clone)]
+    pub struct PubkeyHasherBuilder {
+        offset: usize,
+    }
+
+    impl Default for PubkeyHasherBuilder {
+        fn default() -> Self {
+            let mut rng = rand::prelude::thread_rng();
+            PubkeyHasherBuilder {
+                offset: rng.gen_range(0..32 - 8),
```

Comment:
This is probably too expensive to do for each hashmap. We should do what RandomState does, which is to generate a TLS offset once.

Reply:
Great idea, done in [333a0ed](https://github.com/anza-xyz/solana-sdk/pull/96/commits/333a0ed3c1398b19e44136b130646ad52f0dbb66)